### PR TITLE
Create red_black_tree.cpp

### PR DESCRIPTION
**Pull request by yatharthshahjpr**
code for red_black_tree using c++

### CODE REVIEW

**Comment by AdiChat** [comment]
Looks good 👍 

Hope you are enjoying your journey with **OpenGenus**

File: code/data_structures/red_black_tree/red_black_tree.cpp [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+{
+    int data;
+    bool color;
+    Node *left, *right, *parent;
+ 
+    // Constructor
+    Node(int data)
+    {
+       this->data = data;
+       left = right = parent = NULL;
```
Comment (arnavb) [inline code/data_structures/red_black_tree/red_black_tree.cpp]: Use `nullptr`, not `NULL`, this is C++.

File: code/data_structures/red_black_tree/red_black_tree.cpp [hunk 2]
```diff
@@ -0,0 +1,272 @@
+#include <bits/stdc++.h>
+using namespace std;
+ 
+enum Color {RED, BLACK};
```
Comment (arnavb) [inline code/data_structures/red_black_tree/red_black_tree.cpp]: use `enum class` instead.

File: code/data_structures/red_black_tree/red_black_tree.cpp [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+{
+private:
+    Node *root;
+protected:
+    void rotateLeft(Node *&, Node *&);
+    void rotateRight(Node *&, Node *&);
+    void fixViolation(Node *&, Node *&);
+public:
+    // Constructor
+    RBTree() { root = NULL; }
```
Comment (arnavb) [inline code/data_structures/red_black_tree/red_black_tree.cpp]: Same here; use `nullptr`

File: code/data_structures/red_black_tree/red_black_tree.cpp [hunk 4]
```diff
@@ -0,0 +1,272 @@
...
+ 
+    tree.insert(7);
+    tree.insert(6);
+    tree.insert(5);
+    tree.insert(4);
+    tree.insert(3);
+    tree.insert(2);
+    tree.insert(1);
+ 
+    cout << "Inoder Traversal of Created Tree\n";
```
Comment (arnavb) [inline code/data_structures/red_black_tree/red_black_tree.cpp]: Typo, inoder -> inorder.

**Comment by arnavb** [comment]
Oh wait, my review was too late. R.I.P. 😢

**Comment by AdiChat** [comment]
Oh. Do not worry 😄

**Pull request merged**